### ISSUE TITLE
Add HTML output element for dynamic text

### ISSUE THREAD

**Issue by dqnykamp** [Doenet/DoenetML#632]
The HTML [output element](https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/output) can be used to demarcate which text will change dynamically based on which inputs.

We should explore how we could determine what inputs or other dynamic elements a given DoenetML component is dependent on with the goal of automatically inserting HTML output elements that correctly describe when the contents of the component could change. In the worst case scenario, we could add a DoenetML component to specify this relationship, but it'd be much better if the output elements were added with no additional author effort required.

We do have the directed graph of dependencies among components, so this task is theoretically possible.

It's also not clear what the best practices are on the amount of content that should be wrapped by the output element. For example, if a word changes dynamically, should just the word or the whole paragraph be wrapped?
